### ISSUE TITLE
[no-empty-pattern] gets triggered on a non-destructuring array declaration

### ISSUE THREAD

**Issue by mcuelenaere** [bradzacher/eslint-plugin-typescript#155]
Minimal testcase that should reproduce the error:

`.eslintrc.json`:
```json
{
  "extends": [
    "typescript"
  ]
}
```

`test.js`:
```javascript
const x = [];
const y = x.reduce(a => a, []).indexOf('abc') !== -1;
```

`eslint test.js`:
```shell

/Users/mcuelenaere/test.js
  2:28  error  Unexpected empty array pattern  no-empty-pattern

✖ 1 problem (1 error, 0 warnings)
```
